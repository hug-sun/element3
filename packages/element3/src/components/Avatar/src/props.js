import { isString, isNumber } from '../../../utils/types'

export const props = {
  icon: {
    type: String
  },
  size: {
    type: [Number, String],
    default: 'large',
    validator(val) {
      if (isString(val)) {
        return ['large', 'medium', 'small'].includes(val)
      }
      return isNumber(val)
    }
  },
  shape: {
    type: String,
    default: 'circle',
    validator(val) {
      return ['circle', 'square'].includes(val)
    }
  },
  src: {
    type: String
  },
  alt: {
    type: String
  },
  srcSet: {
    type: String
  },
  fit: {
    type: String,
    default: 'cover'
  },
  error: {
    type: Function
  }
}
