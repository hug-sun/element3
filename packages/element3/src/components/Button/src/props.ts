import { ButtonType, ButtonSize, ButtonNativeType } from './types'
import { PropType } from 'vue'

export const props = {
  size: {
    type: String as PropType<ButtonSize>,
    validator(val: string): boolean {
      return ['medium', 'small', 'mini', ''].includes(val)
    }
  },
  type: {
    type: String as PropType<ButtonType>,
    validator(val: string): boolean {
      return [
        'primary',
        'success',
        'warning',
        'danger',
        'info',
        'text'
      ].includes(val)
    }
  },
  nativeType: {
    type: String as PropType<ButtonNativeType>,
    default: 'button'
  },
  plain: Boolean,
  round: Boolean,
  circle: Boolean,
  loading: Boolean,
  disabled: Boolean,
  icon: String
}
