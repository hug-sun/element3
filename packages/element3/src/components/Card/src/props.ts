import { CardShadow } from './types'
import { PropType } from 'vue'

export const props = {
  header: String,
  bodyStyle: {
    type: Object,
    default: () => {
      return {
        padding: '20px'
      }
    }
  },
  shadow: {
    type: String as PropType<CardShadow>,
    default: 'always',
    validator(val: string): boolean {
      return ['always', 'hover', 'never'].includes(val)
    }
  }
}
