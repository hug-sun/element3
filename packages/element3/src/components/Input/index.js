import ElInput from './src/Input.vue'

/* istanbul ignore next */
ElInput.install = function (app) {
  app.component(ElInput.name, ElInput)
}

export { ElInput }
