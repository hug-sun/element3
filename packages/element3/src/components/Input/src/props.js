export const props = {
  showWordLimit: {
    type: Boolean,
    default: false
  },
  modelValue: {
    type: [String, Number],
    default: ''
  },
  clearable: {
    type: Boolean,
    default: false
  },
  showPassword: {
    type: Boolean,
    default: false
  },
  prefixIcon: {
    type: String
  },
  suffixIcon: {
    type: String
  },
  validateEvent: {
    type: Boolean,
    default: true
  },
  type: {
    type: String,
    default: 'text'
  },
  size: {
    type: String,
    validator: function (value) {
      return ['medium', 'small', 'mini', ''].includes(value)
    }
  },
  autosize: {
    type: [Boolean, Object],
    default: false
  },
  resize: {
    type: String
  }
}
