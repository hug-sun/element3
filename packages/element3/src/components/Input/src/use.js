import { ref, computed, watch, reactive, toRefs, inject, onMounted } from 'vue'
import calcTextareaHeight from './calcTextareaHeight'

export const useClass = (inputSize, inputExceed, props, attrs, slots) => {
  return computed(() => {
    const classes = [
      inputSize.value ? 'el-input--' + inputSize.value : '',
      props.type === 'textarea' ? 'el-textarea' : 'el-input'
    ]

    if (attrs.disabled) {
      classes.push('is-disabled')
    }

    if (inputExceed.value) {
      classes.push('is-exceed')
    }

    if (slots.prepend || slots.append) {
      classes.push('el-input-group')
    }

    if (slots.append) {
      classes.push('el-input-group--append')
    }

    if (slots.prepend) {
      classes.push('el-input-group--prepend')
    }
    if (slots.prefix || props.prefixIcon) {
      classes.push('el-input--prefix')
    }

    if (
      slots.suffix ||
      props.suffixIcon ||
      slots.suffixIcon ||
      attrs.clearable ||
      attrs.showPassword
    ) {
      classes.push('el-input--suffix')
    }

    return classes
  })
}

export const useInput = (props, cxt, textarea) => {
  const input = ref(null)

  const elFormItem = inject('elFormItem', {})

  const elFormChange = inject('elForm.change', () => {})

  const {
    modelValue,
    size,
    suffixIcon,
    clearable,
    showPassword,
    showWordLimit
  } = toRefs(props)

  const nativeInputValue = computed(() => {
    return modelValue.value === null || modelValue.value === undefined
      ? ''
      : String(modelValue.value)
  })

  const textLength = computed(() => {
    if (typeof modelValue.value === 'number') {
      return String(modelValue.value).length >= Number(cxt.attrs.maxlength)
        ? Number(cxt.attrs.maxlength)
        : String(modelValue.value).length
    }

    return modelValue.value.length >= Number(cxt.attrs.maxlength)
      ? Number(cxt.attrs.maxlength)
      : modelValue.value.length
  })

  const elFormItemSize = computed(() => {
    return elFormItem.elFormItemSize || ''
  })

  const inputSize = computed(() => {
    return size?.value || elFormItemSize.value
  })

  const inputExceed = computed(() => {
    return props.modelValue?.length >= Number(cxt.attrs.maxlength)
      ? true
      : false
  })

  const getInput = () =>
    computed({
      get: () => (input.value ? input.value : textarea.value),
      set: (value) => {
        if (input?.value) {
          input.value.value = value
        }

        if (textarea?.value) {
          textarea.value.value = value
        }
      }
    })

  const clearValue = () => {
    cxt.emit('update:modelValue', '')
  }

  const setNativeInputValue = () => {
    const input = getInput()
    if (!input) return
    input.value = nativeInputValue.value
  }

  const getSuffixVisible = computed(() => {
    return (
      cxt.slots.suffix ||
      suffixIcon?.value ||
      clearable?.value ||
      showPassword?.value ||
      showWordLimit?.value
    )
  })

  onMounted(() => {
    setNativeInputValue(nativeInputValue.value)
  })

  watch(
    () => props.modelValue,
    () => {
      setNativeInputValue()
      if (props.validateEvent) {
        elFormChange()
      }
    }
  )

  return {
    input,
    getInput,
    nativeInputValue,
    textLength,
    clearValue,
    inputSize,
    getSuffixVisible,
    inputExceed
  }
}

export const useInputEvent = (emit) => {
  const handleInput = (event) => {
    emit('update:modelValue', event.target.value)
    emit('input', event.target.value)
  }

  const handleFocus = (event) => {
    emit('focus', event)
  }

  const handleBlur = (event) => emit('blur', event)

  const handleClear = () => {
    emit('update:modelValue', '')
    emit('clear')
  }

  const onChange = (event) => {
    emit('change', event.target.value)
  }

  return {
    handleInput,
    handleFocus,
    handleBlur,
    handleClear,
    onChange
  }
}

export const useInputMethod = (element) => {
  const focus = () => {
    element.value.focus()
  }

  const blur = () => {
    element.value.blur()
  }

  const select = () => {
    element.value.select()
  }

  return {
    focus,
    blur,
    select
  }
}

export const useTextarea = (props) => {
  const textarea = ref(null)

  const { autosize, type, resize } = toRefs(props)

  const state = reactive({
    textareaCalcStyle: {}
  })

  const textareaStyle = computed(() => {
    return Object.assign({}, state.textareaCalcStyle, { resize: resize?.value })
  })

  watch(
    () => props.modelValue,
    () => resizeTextarea()
  )

  const resizeTextarea = () => {
    if (type.value !== 'textarea') return
    if (!autosize.value) {
      state.textareaCalcStyle = {
        minHeight: calcTextareaHeight(textarea.value).minHeight
      }

      return
    }
    const minRows = autosize.value.minRows
    const maxRows = autosize.value.maxRows
    state.textareaCalcStyle = calcTextareaHeight(
      textarea.value,
      minRows,
      maxRows
    )
  }

  return { textarea, textareaStyle, resizeTextarea }
}
