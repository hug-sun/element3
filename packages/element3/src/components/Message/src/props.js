export const props = {
  message: {
    type: [String, Object]
  },
  visiable: {
    type: Boolean,
    default: true
  },
  type: {
    type: String,
    defalut: 'info',
    validator(val) {
      return ['success', 'warning', 'info', 'error'].includes(val)
    }
  },
  iconClass: String,
  showClose: Boolean,
  duration: Number,
  center: Boolean,
  customClass: String,
  dangerouslyUseHTMLString: Boolean,
  offset: Number
}
