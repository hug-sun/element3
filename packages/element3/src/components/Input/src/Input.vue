<template>
  <div :style="$attrs.style" :class="classes">
    <template v-if="type !== 'textarea'">
      <div class="el-input-group__prepend" v-if="$slots.prepend">
        <slot name="prepend"></slot>
      </div>
      <input
        class="el-input__inner"
        ref="input"
        v-bind="$attrs"
        @blur="handleBlur"
        @focus="handleFocus"
        @change="onChange"
        :type="showPassword ? (isVisiablePassword ? 'text' : 'password') : type"
        @input="handleInput"
      />
      <span class="el-input__prefix" v-if="$slots.prefix || prefixIcon">
        <slot name="prefix"></slot>
        <i class="el-input__icon" v-if="prefixIcon" :class="prefixIcon"></i>
      </span>
      <span class="el-input__suffix" v-if="getSuffixVisible">
        <span class="el-input__suffix-inner">
          <template v-if="!clearable || !showPassword || !showWordLimit">
            <slot name="suffix"></slot>
            <i class="el-input__icon" v-if="suffixIcon" :class="suffixIcon"></i>
          </template>
          <i
            v-if="clearable"
            class="el-input__icon el-icon-circle-close el-input__clear"
            @mousedown.prevent
            @click.prevent="handleClear"
          ></i>

          <i
            v-if="showPassword"
            class="el-input__icon el-icon-view el-input__clear"
            @mousedown.prevent
            @click.prevent="togglePassword"
          ></i>

          <span v-if="showWordLimit" class="el-input__count">
            <span class="el-input__count-inner"
              >{{ textLength }}/{{ $attrs.maxlength }}</span
            >
          </span>
        </span>
      </span>
      <!-- 后置元素 -->
      <div class="el-input-group__append" v-if="$slots.append">
        <slot name="append"></slot>
      </div>
    </template>
    <template v-else>
      <textarea
        class="el-textarea__inner"
        ref="textarea"
        :style="textareaStyle"
        v-bind="$attrs"
        @input="handleInput"
        @blur="handleBlur"
        @focus="handleFocus"
        @change="onChange"
      >
      </textarea>
      <span v-if="showWordLimit" class="el-input__count"
        >{{ modelValue.length }}/{{ $attrs.maxlength }}</span
      >
    </template>
  </div>
</template>
<script>
import { defineComponent, toRef, reactive } from 'vue'
import { props } from './props'

import {
  useInput,
  useTextarea,
  useInputMethod,
  useInputEvent,
  useClass
} from './use'

export default defineComponent({
  name: 'ElInput',
  inheritAttrs: false,
  props: props,
  emits: ['blur', 'focus', 'change', 'input', 'clear', 'update:modelValue'],

  setup(props, cxt) {
    const state = reactive({
      isVisiablePassword: false
    })

    const { inputSize } = useInput(props, cxt)

    const { attrs, emit } = cxt

    const { textarea, textareaStyle } = useTextarea(props)

    const {
      input,
      textLength,
      getSuffixVisible,
      inputExceed,
      getInput
    } = useInput(props, cxt, textarea)

    const classes = useClass(
      inputSize,
      inputExceed,
      props,
      cxt.attrs,
      cxt.slots
    )
    const {
      handleInput,
      handleFocus,
      handleBlur,
      handleClear,
      onChange
    } = useInputEvent(emit)

    const { focus, select, blur } = useInputMethod(getInput())

    const togglePassword = () =>
      (state.isVisiablePassword = !state.isVisiablePassword)

    return {
      isVisiablePassword: toRef(state, 'isVisiablePassword'),
      focus,
      select,
      blur,
      handleBlur,
      handleInput,
      handleFocus,
      togglePassword,
      onChange,
      getSuffixVisible,
      attrs,
      input,
      textarea,
      handleClear,
      textLength,
      textareaStyle,
      classes
    }
  }
})
</script>
