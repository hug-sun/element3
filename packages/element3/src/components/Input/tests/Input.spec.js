import Input from '../src/Input.vue'
import { mount, shallowMount } from '@vue/test-utils'
import { nextTick, ref } from 'vue'
import { props } from '../src/props'

describe('test props size valiate', () => {
  it('snapshot', () => {
    const wrapper = mount(Input)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('test props size valiate ', () => {
    expect(props.size.validator('mini')).toBeTruthy()
    expect(props.size.validator('small')).toBeTruthy()
    expect(props.size.validator('medium')).toBeTruthy()
  })
})
describe('render children by props.type', () => {
  it('should render input text ', async () => {
    const wrapper = shallowMount(Input, {
      propsData: {
        type: 'button'
      }
    })

    expect(wrapper.find('input').isVisible()).toBe(true)
    expect(wrapper.find('textarea').exists()).toBe(false)
  })

  it('should render input textarea ', async () => {
    const wrapper = mount(Input, {
      propsData: {
        type: 'textarea'
      }
    })

    expect(wrapper.find('textarea').isVisible()).toBe(true)
    expect(wrapper.find('input').exists()).toBe(false)
  })
})

describe('props', () => {
  it('should render disable', async () => {
    const wrapper = mount(Input, {
      attrs: {
        disabled: 'disabled'
      }
    })
    await nextTick()
    expect(wrapper.find('.el-input__inner').attributes()).toHaveProperty(
      'disabled'
    )
  })
})

describe('show-word-limit', () => {
  it('should render limit-word', async () => {
    const wrapper = mount({
      data: () => {
        return {
          text: 'text'
        }
      },
      components: { Input },
      template: `<Input type="text" v-model="text" maxlength=10 show-word-limit />
        >`
    })

    expect(wrapper.find('.el-input').classes()).not.toContain('is-exceed')
    wrapper.find('.el-input__inner').setValue('content test')

    await nextTick()
    expect(wrapper.find('.el-input__count-inner').text()).toContain(`10/10`)
    expect(wrapper.find('.el-input').classes()).toContain('is-exceed')
  })
})

describe('clearable', () => {
  it('should render clear btn', () => {
    const wrapper = mount({
      setup() {
        const value = ref('')
        return {
          value
        }
      },
      components: { Input },
      template: `<Input v-model="value"  clearable/>`
    })

    expect(wrapper.find('.el-input__clear').exists()).toBeTruthy()
  })

  it('clear value when click clear btn', async () => {
    const wrapper = mount({
      setup() {
        const input = ref('value')
        return {
          input
        }
      },
      components: { Input },
      template: `<Input v-model="input" type="text" clearable/>`
    })

    wrapper.find('.el-input__clear').trigger('click')
    await nextTick()
    expect(wrapper.vm.input).toBe('')
  })
})

describe('password', () => {
  it('show password input', async () => {
    const wrapper = mount({
      data: () => {
        return {
          value: 'test'
        }
      },

      components: { Input },
      template: `<Input v-model="value" type="password" showPassword/>`
    })

    wrapper.find('.el-icon-view').trigger('click')

    await nextTick()
    expect(wrapper.find('.el-input__inner').attributes('type')).toEqual('text')

    wrapper.find('.el-icon-view').trigger('click')

    await nextTick()
    expect(wrapper.find('.el-input__inner').attributes('type')).toEqual(
      'password'
    )
  })
})

describe('size', () => {
  it('should render props.size', () => {
    const wrapper = mount(Input, {
      propsData: {
        size: 'medium'
      }
    })

    expect(wrapper.find('div').classes()).toContain('el-input--medium')
  })
})

describe('prefixIcon', () => {
  it('should render props.prefixIcon', () => {
    const wrapper = mount(Input, {
      propsData: {
        prefixIcon: 'el-icon-view'
      }
    })

    expect(wrapper.find('.el-input__prefix').exists()).toBe(true)
  })
})

describe('suffix-icon', () => {
  it('should render props.suffixIcon', () => {
    const wrapper = mount(Input, {
      propsData: {
        suffixIcon: 'el-icon-view'
      }
    })

    expect(wrapper.find('.el-input__suffix').exists()).toBe(true)
  })
})

describe('autocomplete', () => {
  it('should render autocomplete', async () => {
    const wrapper = mount(Input, {
      attrs: {
        autocomplete: 'off'
      }
    })

    expect(wrapper.find('.el-input__inner').attributes('autocomplete')).toEqual(
      'off'
    )
  })
})

describe('name attribute', () => {
  it('should render name attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        name: 'username'
      }
    })

    expect(wrapper.find('.el-input__inner').attributes('name')).toEqual(
      'username'
    )
  })
})

describe('readonly attribute', () => {
  it('should render readonly attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        readonly: ''
      }
    })

    expect(wrapper.find('.el-input__inner').attributes()).toHaveProperty(
      'readonly'
    )
  })
})

describe('max attribute', () => {
  it('should render max attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        max: 13
      }
    })

    expect(wrapper.find('.el-input__inner').attributes('max')).toEqual('13')
  })
})

describe('min attribute', () => {
  it('should render min attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        min: 13
      }
    })

    expect(wrapper.find('.el-input__inner').attributes('min')).toEqual('13')
  })
})

describe('step attribute', () => {
  it('should render step attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        step: ','
      }
    })

    expect(wrapper.find('.el-input__inner').attributes('step')).toEqual(',')
  })
})

describe('autofocus attribute', () => {
  it('should render autofocus attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        autofocus: 'autofocus'
      }
    })

    expect(wrapper.find('.el-input__inner').attributes()).toHaveProperty(
      'autofocus'
    )
  })
})

describe('form attribute', () => {
  it('should render form attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        form: 'form_id'
      }
    })

    expect(wrapper.find('.el-input__inner').attributes('form')).toEqual(
      'form_id'
    )
  })
})

describe('label attribute', () => {
  it('should render label attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        label: 'label'
      }
    })

    expect(wrapper.find('.el-input__inner').attributes('label')).toEqual(
      'label'
    )
  })
})

describe('tabindex attribute', () => {
  it('should render tabindex attribute', () => {
    const wrapper = mount(Input, {
      attrs: {
        tabindex: 12
      }
    })

    expect(wrapper.find('.el-input__inner').attributes('tabindex')).toEqual(
      '12'
    )
  })
})

describe('validate-event attribute', () => {
  it('should call validate-event when validate-event eq true', async () => {
    var called = false
    const wrapper = mount(Input, {
      propsData: {
        modelValue: 'test',
        validateEvent: true
      },
      global: {
        provide: {
          'elForm.change': () => {
            called = true
          }
        }
      }
    })

    await nextTick()

    wrapper.setProps({ modelValue: 'content' })
    await nextTick()
    expect(called).toBeTruthy()
  })

  it('should call validate-event when validate-event eq false', async () => {
    var called = false
    const wrapper = mount(Input, {
      propsData: {
        modelValue: 'test',
        validateEvent: false
      },
      global: {
        provide: {
          'elForm.change': () => {
            called = true
          }
        }
      }
    })

    await nextTick()

    wrapper.setProps({ modelValue: 'content' })
    await nextTick()
    expect(called).toBeFalsy()
  })
})

describe('solt', () => {
  it('should render solt prefix', async () => {
    const slot = 'prefix solt'
    const wrapper = mount(Input, {
      propsData: {
        prefixIcon: 'el-view'
      },
      slots: {
        prefix: slot
      }
    })

    await nextTick()
    expect(wrapper.find('.el-input__prefix').html()).toContain(slot)
  })

  it('should render solt suffix', async () => {
    const slot = 'suffix solt'
    const wrapper = mount(Input, {
      propsData: {
        suffixIcon: 'el-view'
      },
      slots: {
        suffix: slot
      }
    })

    await nextTick()
    expect(wrapper.find('.el-input__suffix').html()).toContain(slot)
  })

  it('should render solt prepend', async () => {
    const slot = 'prepend solt'
    const wrapper = mount(Input, {
      slots: {
        prepend: slot
      }
    })

    await nextTick()
    expect(wrapper.find('.el-input-group__prepend').html()).toContain(slot)
  })

  it('should render solt append', async () => {
    const slot = 'append solt'
    const wrapper = mount(Input, {
      slots: {
        append: slot
      }
    })

    await nextTick()
    expect(wrapper.find('.el-input-group__append').html()).toContain(slot)
  })
})

describe('blur', () => {
  it('should call custome blur event', async () => {
    const wrapper = mount(Input)

    wrapper.find('input').trigger('blur')
    expect(wrapper.emitted()).toHaveProperty('blur')
  })
})

describe('focus', () => {
  it('should call custome focus event', async () => {
    const wrapper = mount(Input)

    wrapper.find('input').trigger('focus')
    expect(wrapper.emitted()).toHaveProperty('focus')
  })
})

describe('change', () => {
  it('should call custome change event', async () => {
    const wrapper = mount(Input, {
      propsData: {
        modelValue: 'value'
      }
    })

    wrapper.find('input').trigger('change')

    expect(wrapper.emitted('change')).toBeTruthy()
    expect(wrapper.emitted('change')[0][0]).toEqual('value')
  })
})

describe('input', () => {
  it('should call custome input event', async () => {
    const wrapper = mount(Input, {
      propsData: {
        modelValue: 'value'
      }
    })

    wrapper.find('input').trigger('input')

    expect(wrapper.emitted('input')).toBeTruthy()
    expect(wrapper.emitted('input')[0][0]).toEqual('value')
  })
})
describe('clear', () => {
  it('click clear btn should call custom call', () => {
    const wrapper = mount(Input, {
      propsData: {
        modelValue: 'value',
        suffixIcon: 'el-view'
      },
      attrs: {
        clearable: true
      }
    })
    wrapper.find('.el-input__clear').trigger('click')

    expect(wrapper.emitted('clear')).toBeTruthy()
  })
})

describe('methods', () => {
  it('should export method focus', () => {
    const wrapper = mount({
      setup() {
        const inputRef = ref(null)
        return {
          inputRef
        }
      },
      components: { Input },
      template: `<Input ref="inputRef" />`
    })

    expect(wrapper.vm.inputRef.focus).toBeDefined()
  })

  it('should export method blur', () => {
    const wrapper = mount({
      setup() {
        const inputRef = ref(null)
        return {
          inputRef
        }
      },
      components: { Input },
      template: `<Input ref="inputRef"/>`
    })

    expect(wrapper.vm.inputRef.blur).toBeDefined()
  })

  it('should export method select', () => {
    const wrapper = mount({
      setup() {
        const inputRef = ref(null)
        return {
          inputRef
        }
      },
      components: { Input },
      template: `<Input ref="inputRef"/>`
    })

    expect(wrapper.vm.inputRef.select).toBeDefined()
  })
})
