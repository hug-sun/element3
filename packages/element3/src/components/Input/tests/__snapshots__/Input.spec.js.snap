// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test props size valiate snapshot 1`] = `
<div
  class="el-input"
>
  
  <!--v-if-->
  <input
    class="el-input__inner"
    type="text"
  />
  <!--v-if-->
  <!--v-if-->
  <!-- 后置元素 -->
  <!--v-if-->
  
</div>
`;
